alibi_detect/od/sr.py

```diff
@@ -1,19 +1,37 @@
+import enum
 import logging
 import numpy as np
 from typing import Dict
 from alibi_detect.base import BaseDetector, ThresholdMixin, outlier_prediction_dict
+from alibi_detect.utils._types import Literal
 
 logger = logging.getLogger(__name__)
 
+# numerical stability for division
 EPSILON = 1e-8
 
 
+class Padding(str, enum.Enum):
+    CONSTANT = 'constant'
+    REPLICATE = 'replicate'
+    REFLECT = 'reflect'
+
+
+class Side(str, enum.Enum):
+    BILATERAL = 'bilateral'
+    LEFT = 'left'
+    RIGHT = 'right'
+
+
 class SpectralResidual(BaseDetector, ThresholdMixin):
 
     def __init__(self,
                  threshold: float = None,
                  window_amp: int = None,
                  window_local: int = None,
+                 padding_amp_method: Literal['constant', 'replicate', 'reflect'] = 'reflect',
```

Comment:
Does it make sense to have different padding methods for `amp` and `local`?

Reply:
It can definitely be reduced to a single padding method that would be applied in both time and frequency domain. 
It was designed with separate padding methods for benchmarking purposes.  The `'reflect'` padding performed best in my experiments for both domains.